base/dcp_client_metadata.go

```diff
@@ -159,19 +159,19 @@ func BuildDCPMetadataSliceFromVBUUIDs(vbUUIDS []uint64) []DCPMetadata {
 // DCPMetadataCS stores DCP metadata in the specified CouchbaseStore.  It does not require that the store is the
 // same one being streamed over DCP.
 type DCPMetadataCS struct {
-	bucket    Bucket
-	keyPrefix string
-	metadata  []DCPMetadata
-	endSeqNos []gocbcore.SeqNo
+	collection *Collection
```

Comment:
Does DCPMetadataCS actually require a *Collection?  I don't see anywhere we were previously casting to *Collection in the implementation.  If it actually just requires Bucket, that makes it more easily unit testable (walrus).

Reply:
It puts data into a couchbase store, so I suppose we can leave it as a bucket. It's not tested under walrus though otherwise this code would fail.